pkg/services/librarypanels/database.go

```diff
@@ -365,60 +366,101 @@ func (lps *LibraryPanelService) getLibraryPanel(c *models.ReqContext, uid string
 }
 
 // getAllLibraryPanels gets all library panels.
-func (lps *LibraryPanelService) getAllLibraryPanels(c *models.ReqContext, limit int64) ([]LibraryPanelDTO, error) {
+func (lps *LibraryPanelService) getAllLibraryPanels(c *models.ReqContext, perPage int, page int, name string, excludeUID string) (LibraryPanelSearchResult, error) {
 	libraryPanels := make([]LibraryPanelWithMeta, 0)
+	result := LibraryPanelSearchResult{}
+	if perPage <= 0 {
+		perPage = 100
+	}
+	if page <= 0 {
+		page = 1
+	}
+
 	err := lps.SQLStore.WithDbSession(c.Context.Req.Context(), func(session *sqlstore.DBSession) error {
 		builder := sqlstore.SQLBuilder{}
 		builder.Write(sqlStatmentLibrayPanelDTOWithMeta)
 		builder.Write(` WHERE lp.org_id=? AND lp.folder_id=0`, c.SignedInUser.OrgId)
+		if len(strings.TrimSpace(name)) > 0 {
+			builder.Write(" AND lp.name "+lps.SQLStore.Dialect.LikeStr()+" ?", "%"+name+"%")
```

Comment:
This looks like it's vulnerable to SQL injection, think in other places we use " LIKE ?" and pass in a string as the param, https://github.com/grafana/grafana/blob/master/pkg/services/sqlstore/org.go#L32

Reply:
Good catch, I was inspired by the code here:
https://github.com/grafana/grafana/blob/5bb1b1602d1e5bd614e2c11fa3a1e984df988357/pkg/services/sqlstore/team.go#L202 but I also found this place:
https://github.com/grafana/grafana/blob/5bb1b1602d1e5bd614e2c11fa3a1e984df988357/pkg/services/sqlstore/alert.go#L96

Not sure I understand what the difference is though between this and the code in https://github.com/grafana/grafana/blob/master/pkg/services/sqlstore/org.go#L32. Is it because I'm using the builder here and in `org.go` is using `Table().Where()`. Basically I think I'm missing some context here about the inner workings of `xorm`.

Reply:
Sorry! was me who read the code wrong, I missed the ? and the fact that the name was supplied as an argument to Write. See now that Write handled the second argument as proper parameter . 
https://github.com/grafana/grafana/blob/5bb1b1602d1e5bd614e2c11fa3a1e984df988357/pkg/services/sqlstore/sqlbuilder.go#L19